src/Http/Http.Features/src/IHttpWebTransportFeature.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.Versioning;
+
+namespace Microsoft.AspNetCore.Http.Features;
+
+/// <summary>
+/// API for accepting and retrieving WebTransport sessions.
+/// </summary>
+public interface IHttpWebTransportFeature
+{
+    /// <summary>
+    /// Indicates if this request is a WebTransport request.
+    /// </summary>
+    bool IsWebTransportRequest { get; }
+
+    /// <summary>
+    /// Accept the session request and allow streams to start being used.
+    /// </summary>
+    /// <param name="cancellationToken">The cancellation token to cancel waiting for the session.</param>
+    /// <returns>An instance of a WebTransportSession which will be used to control the connection.</returns>
+    [RequiresPreviewFeatures("WebTransport is a preview feature")]
```

Comment:
Yes, but unfortunately, as this annotation recurses up through the callers, this means that either a) almost all of kestrel needs to be marked as requires preview features, or b) we have a ton of `#pragma warning disable CA2252` everywhere. To avoid this, Chris and I decided to just place it on the entry point of using WebTransport, i.e. this accept method.